app/scripts/views/sign_in.js

```diff
@@ -37,6 +37,28 @@ define(function (require, exports, module) {
 
     beforeRender: function () {
       this._account = this._suggestedAccount();
+
+      // Extract email to see if valid account
+      var relierEmail = this.relier.get('email');
+      var formEmail = this._formPrefill.get('email');
+
+      // Use the relier's email to check for valid account.
+      // If the form email has been set, this signals that the user has modified the form email input
+      // and we want to use it for later updates
+      if (relierEmail && ! formEmail) {
```

Comment:
IIUC, the desired behavior for this functionality is to send the user to the `/oauth/` page instead of directly to `/oauth/signin`.  I may be mistaken, but if the user is sent to `/oauth/`, this check should be done in [router.js->redirectToBestOAuthChoice](https://github.com/mozilla/fxa-content-server/blob/dfeb53bf0624298c883710c3184c51aba3de5ad8/app/scripts/lib/router.js#L164). The check will be done once, and should be significantly simpler because you won't need to worry about interactions w/ formPrefill, and logic won't be spread across sign_in.js and sign_up.js.